### PR TITLE
Script improvement

### PR DESCRIPTION
**Pull request by khancyr**
Use shellcheck to check all our script
Add double quote on some variable
Remove useless loop on build_ci.sh as we only support waf as build tools now.
Add correct extension to some script

### CODE REVIEW

File: Tools/scripts/build-jsbsim.sh [hunk 1]
```diff
@@ -21,7 +21,7 @@ else
     make -j2
 fi
 
-if [[ ! -n $(echo $PATH | grep jsbsim) ]]; then
+if [[ ! -n $(echo "$PATH" | grep jsbsim) ]]; then
```
Comment (peterbarker) [inline Tools/scripts/build-jsbsim.sh]: Echo echos all arguments, so this is a noop AFAICS.

Still not a bad change.

File: Tools/scripts/build_all.sh [hunk 2]
```diff
@@ -15,7 +15,7 @@ BOARDS="sitl linux Pixhawk1"
 
 for b in $BOARDS; do
     echo "Testing $b build"
-    ./waf configure --board $b
+    ./waf configure --board "$b"
```
Comment (peterbarker) [inline Tools/scripts/build_all.sh]: If we *ever* put a space in a board name then *VAST* numbers of things are going to break :-)

File: Tools/scripts/install-apt-ci.sh [hunk 3]
```diff
@@ -17,12 +17,12 @@ PKGS=" \
...
+read -r UBUNTU_CODENAME <<<"$(lsb_release -c -s)"
 
 sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
 
 #wget -q -O - http://llvm.org/apt/llvm-snapshot.gpg.key | sudo apt-key add -
 #sudo add-apt-repository "deb http://llvm.org/apt/${UBUNTU_CODENAME}/ llvm-toolchain-${UBUNTU_CODENAME}-3.7 main" -y
 sudo apt-get -qq -y --force-yes update
 sudo apt-get -qq -y --force-yes remove clang llvm
-sudo apt-get -y --force-yes install $PKGS
+sudo apt-get -y --force-yes install "$PKGS"
```
Comment (peterbarker) [inline Tools/scripts/install-apt-ci.sh]: Wait, is this correct?!

We *want* it to expand `PKGS` here!

File: Tools/scripts/run-coverage.sh [hunk 4]
```diff
@@ -23,18 +23,18 @@ rm -rf build
 # Run examples
 ./waf configure --board=linux --debug
 ./waf examples
-./Tools/autotest/autotest.py $OPTS run.examples
+./Tools/autotest/autotest.py "$OPTS" run.examples
```
Comment (peterbarker) [inline Tools/scripts/run-coverage.sh]: Did you test this with multiple things in `$OPTS`?

**Comment by khancyr** [comment]
Remove the double quoting stuff for now due to lack of deep testing of all correction

**Comment by khancyr** [comment]
This makes semaphore fails rightfully as some targets don't exist anymore with waf

**Comment by tridge** [comment]
you can't rename dumpstack without fixing the call to it in AP_HAL_SITL

**Comment by khancyr** [comment]
done

**Pull request merged**